src/sensor-fusion/fusion-pose-sensor.js

```diff
@@ -25,211 +30,280 @@ import * as Util from '../util.js';
+    if (Util.isLandscapeMode()) {
+      this.filterToWorldQ.multiply(this.inverseWorldToScreenQ);
+    }
+
+    // Keep track of a reset transform for resetSensor.
+    this.resetQ = new MathUtil.Quaternion();
+
+    this.isFirefoxAndroid = Util.isFirefoxAndroid();
+    this.isIOS = Util.isIOS();
+
+    this.orientationOut_ = new Float32Array(4);
+
+    // Store information on API in use (either 'devicemotion' or 'sensor'),
+    // as well as any errors observed from using sensors for debugging.
+    this.api = null;
+    this.error = null;
+
+    this.onDeviceMotion_ = this.onDeviceMotion_.bind(this);
+    this.onOrientationChange_ = this.onOrientationChange_.bind(this);
+    this.onMessage_ = this.onMessage_.bind(this);
+    this.onSensorError_ = this.onSensorError_.bind(this);
+    this.onGyroscopeRead_ = this.onGyroscopeRead_.bind(this);
+    this.onAccelerometerRead_ = this.onAccelerometerRead_.bind(this);
+
+    this.start();
   }
 
-  this.inverseWorldToScreenQ = new MathUtil.Quaternion();
-  this.worldToScreenQ = new MathUtil.Quaternion();
-  this.originalPoseAdjustQ = new MathUtil.Quaternion();
-  this.originalPoseAdjustQ.setFromAxisAngle(new MathUtil.Vector3(0, 0, 1),
-                                           -window.orientation * Math.PI / 180);
-
-  this.setScreenTransform_();
-  // Adjust this filter for being in landscape mode.
-  if (Util.isLandscapeMode()) {
-    this.filterToWorldQ.multiply(this.inverseWorldToScreenQ);
+  start() {
+    // Only listen for postMessages if we're in an iOS and embedded inside a cross
+    // origin IFrame. In this case, the polyfill can still work if the containing
+    // page sends synthetic devicemotion events. For an example of this, see
+    // the iframe example in the repo at `examples/iframe.html`
+    if (Util.isIOS() && Util.isInsideCrossOriginIFrame()) {
+      window.addEventListener('message', this.onMessage_);
+    }
+
+    window.addEventListener('orientationchange', this.onOrientationChange_);
+
+    // Attempt to use the Gyroscope and Accelerometer from Generic Sensor APIs.
+    // First available in Chrome M63, this can fail for several reasons, and attempt
+    // to fallback to devicemotion. Failure scenarios include:
+    //
+    // * Generic Sensor APIs do not exist, fallback to devicemotion
+    // * Underlying sensor does not exist, no fallback possible.
+    // * Feature policy failure (in an iframe); no fallback.
+    //   * @TODO is it both possible, and respecting of intent, to fallback to
+    //     devicemotion here the same way we handle iframes on iOS?
+    //     @see `onMessage_()`
+    // * Permission to sensor data denied; respect user agent, no fallback to devicemotion
+    let accelerometer = null;
+    let gyroscope = null;
+    try {
+      accelerometer = new Accelerometer({ frequency: SENSOR_FREQUENCY });
+      gyroscope = new Gyroscope({ frequency: SENSOR_FREQUENCY });
+      accelerometer.addEventListener('error', this.onSensorError);
```

Comment:
Is there a need to fuse data manually? Can [RelativeOrientationSensor](https://www.w3.org/TR/orientation-sensor/#relativeorientationsensor-interface) be used? Then, [quaternion](https://www.w3.org/TR/orientation-sensor/#dom-orientationsensor-quaternion) can be obtained without manual fusion. Only screen coordinate sync and other conversions would be needed in getOrientation.

Reply:
That point was already made and he is looking into using that - check the PR comments